rm/rm-server/src/main/java/org/ow2/proactive/resourcemanager/nodesource/infrastructure/InfrastructureManager.java

```diff
@@ -1012,36 +1016,24 @@ public void onDownNodeReconnection(Node node) {
     public RMDeployingNode update(RMDeployingNode rmNode) {
         String nodeUrl = rmNode.getNodeURL();
         RMDeployingNode previousDeployingNode;
-        if (rmNode.isLost() && containsKeyLostNode(nodeUrl)) {
-            previousDeployingNode = putLostNode(nodeUrl, rmNode);
-        } else if (containsKeyDeployingNode(nodeUrl)) {
-            previousDeployingNode = putDeployingNode(nodeUrl, rmNode);
+        if (rmNode.isLost() && containsLostNodeWithLock(nodeUrl)) {
+            previousDeployingNode = addLostNodeWithLockAndPersist(nodeUrl, rmNode);
+        } else if (containsDeployingNodeWithLock(nodeUrl)) {
+            previousDeployingNode = addDeployingNodeWithLockAndPersist(nodeUrl, rmNode);
         } else {
             previousDeployingNode = null;
         }
         return previousDeployingNode;
     }
 
-    void addDeployingNode(RMDeployingNode node) {
-        putDeployingNode(node.getNodeURL(), node);
-    }
-
-    void addLostNode(RMDeployingNode node) {
-        putLostNode(node.getNodeURL(), node);
-    }
-
-    Map<String, RMDeployingNode> getDeployingNodesDeployingState() {
-        return getDeployingNodesMap();
-    }
-
-    Map<String, RMDeployingNode> getDeployingNodesLostState() {
-        return getLostNodesMap();
+    private boolean nodesRecoveryEnabled() {
```

Comment:
![MAJOR](https://sonarsource.github.io/sonar-github/severity-major.png 'Severity: MAJOR') Remove this unused private "nodesRecoveryEnabled" method. [![rule](https://sonarsource.github.io/sonar-github/rule.png)](https://sonarcloud.io/coding_rules#rule_key=squid%3AUnusedPrivateMethod)